source/source.go

```diff
@@ -266,6 +269,25 @@ func suitableType(target string) string {
 	return endpoint.RecordTypeCNAME
 }
 
+// isIPv6String reports whether the target string is an IPv6 address,
+// including IPv4-mapped IPv6 addresses.
+func isIPv6String(target string) bool {
+	netIP, err := netip.ParseAddr(target)
```

Comment:
could you share code coveragate for this functions?

Can we have specific tests just in source_test.go

I know it's a private function, but we may move them at some point to utils.go, this will help with maintenance

Worth to add BenchmarkTest... as well, and if you could, share results